File: Basic Recursion/Check if string is palindrome or not.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    bool palindromeCheck(string& s) {
        return isPalindrome(s, 0, s.length() - 1); 
    }

private:
    bool isPalindrome(string& s, int left, int right) {
        if (left >= right) {
            return true;
        }
        if (s[left] != s[right]) {
            return false;
        }
        return isPalindrome(s, left + 1, right - 1);
    }
};

File: Basic Recursion/Check if string is palindrome or not.py
# Check if String is Palindrome (Recursive)

def is_palindrome(s, left=0, right=None):
    if right is None:
        right = len(s) - 1
    if left >= right:
        return True
    if s[left] != s[right]:
        return False
    return is_palindrome(s, left + 1, right - 1)


if __name__ == "__main__":
    print(is_palindrome("racecar"))  # True
    print(is_palindrome("hello"))    # False
